### PR TITLE
Insert with default propNames

### PR DESCRIPTION
**Pull request by nevermore3**
refer to https://jira.nebula-graph.io/browse/NG-528

### CODE REVIEW

File: tests/tck/features/insert/InsertIfNotExists.feature [hunk 1]
```diff
@@ -168,3 +168,77 @@ Feature: Insert vertex and edge with if not exists
...
+      """
+    Then the execution should be successful
+    When try to execute query:
+      """
+      INSERT VERTEX person VALUES "Tom":("Tom", "2")
+      """
+    Then the execution should be successful
+    When try to execute query:
+      """
+      INSERT VERTEX person VALUES "Tom":("Tom", 2, "3")
```
Comment (Shylock-Hg) [inline tests/tck/features/insert/InsertIfNotExists.feature:217]: The person has two properties only , why can insert three value?

Comment (nevermore3) [reply tests/tck/features/insert/InsertIfNotExists.feature:217]: in process， will be error

File: src/parser/MutateSentences.h [hunk 2]
```diff
@@ -37,49 +37,58 @@ class PropertyList final {
...
-     std::unique_ptr<std::string>               tagName_;
-     std::unique_ptr<PropertyList>              properties_;
+    bool                                       defaultPropNames_{false};
+    std::unique_ptr<std::string>               tagName_;
+    std::unique_ptr<PropertyList>              properties_;
 };
 
 
 class VertexTagList final {
  public:
```
Comment (laura-ding) [inline src/parser/MutateSentences.h:74]: need to delete the space before `public`

File: src/parser/MutateSentences.h [hunk 3]
```diff
@@ -37,49 +37,58 @@ class PropertyList final {
...
+    std::vector<VertexTagItem*> tagItems() const {
+        std::vector<VertexTagItem*> result;
+        result.reserve(tagItems_.size());
+        for (auto &item : tagItems_) {
+            result.emplace_back(item.get());
+        }
+        return result;
+    }
 
  private:
```
Comment (laura-ding) [inline src/parser/MutateSentences.h:90]: ditto

File: src/parser/MutateSentences.h [hunk 4]
```diff
@@ -37,49 +37,58 @@ class PropertyList final {
 
 class VertexTagItem final {
  public:
```
Comment (laura-ding) [inline src/parser/MutateSentences.h:39]: need to delete the space before public

File: src/parser/MutateSentences.h [hunk 5]
```diff
@@ -37,49 +37,58 @@ class PropertyList final {
...
+    }
+
+    std::vector<std::string*> properties() const {
+        if (nullptr == properties_) {
+            return {};
+        }
+        return properties_->properties();
+    }
 
  private:
```
Comment (laura-ding) [inline src/parser/MutateSentences.h:66]: ditto

File: src/validator/MutateValidator.cpp [hunk 6]
```diff
@@ -70,15 +61,23 @@ Status InsertVerticesValidator::check() {
...
-        // Check prop name is in schema
-        for (auto *it : props) {
-            if (schema->getFieldIndex(*it) < 0) {
-                LOG(ERROR) << "Unknown column `" << *it << "' in schema";
-                return Status::SemanticError("Unknown column `%s' in schema", it->c_str());
+        if (item->isDefaultPropNames()) {
+            propSize_ = schema->getNumFields();
+            for (size_t i = 0; i < propSize_; ++i) {
+                const char* propName = schema->getFieldName(i);
+                names.emplace_back(propName);
```
Comment (laura-ding) [inline src/validator/MutateValidator.cpp:68]: the storage supported the request without put the propNames, if empty, the storage will use the schema props.

File: src/parser/MutateSentences.h (hunk 2 above)
Comment (nevermore3) [reply src/parser/MutateSentences.h:74]: done

File: tests/tck/features/insert/InsertIfNotExists.feature [hunk 7]
```diff
@@ -168,3 +168,110 @@ Feature: Insert vertex and edge with if not exists
...
+      """
+      INSERT VERTEX person(name, age) VALUES "Tom":("Tom", 1)
+      """
+    Then the execution should be successful
+    When try to execute query:
+      """
+      INSERT VERTEX person VALUES "Tom":("Tom", 2)
+      """
+    Then the execution should be successful
+    When try to execute query:
```
Comment (yixinglu) [inline tests/tck/features/insert/InsertIfNotExists.feature:195]: Need not to retry when query returns errors, here should be `executing query:`

File: tests/tck/features/insert/InsertIfNotExists.feature [hunk 8]
```diff
@@ -168,3 +168,110 @@ Feature: Insert vertex and edge with if not exists
...
+      """
+    Then the execution should be successful
+    # check edge result with fetch
+    When executing query:
+      """
+      FETCH PROP ON like "Tom"->"Conan" YIELD like.likeness
+      """
+    Then the result should be, in any order:
+      | like._src | like._dst | like._rank | like.likeness |
+      | 'Tom'     | 'Conan'   | 0          | 200           |
```
Comment (yixinglu) [inline tests/tck/features/insert/InsertIfNotExists.feature:277]: Don't forget to drop the tmp space by `And drop the used space`

File: src/validator/MutateValidator.cpp [hunk 9]
```diff
@@ -1,4 +1,4 @@
-/* Copyright (c) 2020 vesoft inc. All rights reserved.
+/* Copyright (c) 2021 vesoft inc. All rights reserved.
```
Comment (Shylock-Hg) [inline src/validator/MutateValidator.cpp:1]: Why change this?

File: tests/tck/features/insert/InsertIfNotExists.feature (hunk 7 above)
Comment (nevermore3) [reply tests/tck/features/insert/InsertIfNotExists.feature]: fix

File: tests/tck/features/insert/InsertIfNotExists.feature (hunk 8 above)
Comment (nevermore3) [reply tests/tck/features/insert/InsertIfNotExists.feature:278]: done

**Review by yixinglu — APPROVED** [review]

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**